src/linagora.esn.unifiedinbox/app/components/sidebar/mailbox-display-tree/mailbox-display-tree.spec.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+/* global chai: false */
+/* global sinon: false */
+
+const { expect } = chai;
+
+describe('The mailboxDisplayTree controller', function() {
+  var $rootScope, scope, $controller, mailboxes, inboxMailboxesService;
+
+  beforeEach(function() {
+    angular.mock.module('linagora.esn.unifiedinbox');
+  });
+
+  beforeEach(function() {
+    mailboxes = [
+      {
+        id: 'id_mailbox',
+        name: 'Mailbox 1',
+        totalMessages: 10,
+        parentId: null,
+        level: 1,
+        qualifiedName: 'Mailbox 1'
+      },
+      {
+        id: 'id_mailbox2',
+        name: 'Mailbox 2',
+        unreadMessages: 0,
+        parentId: 'id_mailbox',
+        level: 2,
+        qualifiedName: 'Mailbox 1 / Mailbox 2'
+      }
+    ];
+
+    angular.mock.inject(function(_$rootScope_, _$controller_, _inboxMailboxesService_) {
+      $rootScope = _$rootScope_;
+      $controller = _$controller_;
+      scope = $rootScope.$new();
+      inboxMailboxesService = _inboxMailboxesService_;
+    });
+  });
+
+  function initController() {
+    var controller = $controller('mailboxDisplayTreeController', {
+      $scope: scope
+    }, {
+      mailboxes: mailboxes,
+      toggle: sinon.spy(),
+      hideBadge: false
+    });
+
+    scope.$digest();
+
+    return controller;
+  }
+
+  it('should call the inboxMailboxesService.mailboxtoTree function', function() {
+    var controller = initController();
+
+    var inboxMailboxesServiceMailboxtoTreeSpy = sinon.spy(inboxMailboxesService, 'mailboxtoTree');
+
+    var mailboxesTree = [
+      {
+        id: 'id_mailbox',
+        name: 'Mailbox 1',
+        totalMessages: 10,
+        parentId: null,
+        level: 1,
+        qualifiedName: 'Mailbox 1',
+        nodes: [
+          {
+            id: 'id_mailbox2',
+            name: 'Mailbox 2',
+            unreadMessages: 0,
+            parentId: 'id_mailbox',
+            level: 2,
+            qualifiedName: 'Mailbox 1 / Mailbox 2'
+          }
```

Comment:
Can you test what happens when a mailbox name contains '/'?

Reply:
Currently, the logic is really simple and add each time between the parent name and the child name `/` but this is now used only for to display flat folder.